libs/CCAppCommon/include/ccApplicationBase.h

```diff
@@ -23,16 +23,33 @@
 
 //! Mimic Qt's qApp for easy access to the application instance
 #define ccApp (static_cast<ccApplicationBase *>( QCoreApplication::instance() ))
-
+//! ccApplicationBase
+/*!
+ * ccApplicationBase provides a QApplication class with additional Constructor param isCommandLine and version
+ * The class also initialises openGL before instantiating the application class
+ * \see QApplication
+ */
 class CCAPPCOMMON_LIB_API ccApplicationBase : public QApplication
 {
 public:
 	//! This must be called before instantiating the application class so it
 	//! can setup OpenGL first.
 	static void	initOpenGL();
-	
+
+    //! ccApplicationBase Constructor
+	/*!
+	 * @param argc argc must be greater than zero, this is set by QT
+	 * @param argv this contains the command line values, see QApplication, again this is set by QT
+	 * @param isCommandLine boolean to define the application as either a Commandline or GUI. If true its a commandline application
+	 * @param version string of the application version set by CloudCompare viewer or app.
+	 */
 	ccApplicationBase( int &argc, char **argv, bool isCommandLine, const QString &version );
-	
+
+    //! isCommandLine
```

Comment:
Simply
`//! Returns whether this application is in command line mode or not`

(should be sufficient, getters might not need a @return field since it's generally redundant with the short description)